darts/timeseries.py

```diff
@@ -140,156 +582,251 @@ def pd_dataframe(self, copy=True) -> pd.DataFrame:
         Returns
         -------
         pandas.DataFrame
-            The Pandas Dataframe underlying this time series
+            The Pandas DataFrame representation of this time series
         """
+        self._assert_deterministic()
         if copy:
-            return self._df.copy()
+            return pd.DataFrame(self._xa[:, :, 0].values.copy(),
+                                index=self._time_index.copy(),
+                                columns=self._xa.get_index('component').copy())
         else:
-            return self._df
+            return pd.DataFrame(self._xa[:, :, 0].values,
+                                index=self._time_index,
+                                columns=self._xa.get_index('component'))
 
-    def columns(self):
-        return self.pd_dataframe().columns
+    def quantile_df(self, quantile=0.5) -> pd.DataFrame:
+        """
+        Returns a Pandas DataFrame containing the single desired quantile of each component (over the samples).
+        Each of the series components will appear as a column in the DataFrame. The column will be named
+        "<component>_X", where "<component>" is the column name corresponding to this component, and "X"
+        is the quantile value.
+        The quantile columns represent the marginal distributions of the components of this series.
+
+        This works only on stochastic series (i.e., with more than 1 sample)
+
+        Parameters
+        ----------
+        quantile
+            The desired quantile value. The value must be represented as a fraction
+            (between 0 and 1 inclusive). For instance, `0.5` will return a DataFrame
+            containing the median of the (marginal) distribution of each component.
+
+        Returns
+        -------
+        pandas.DataFrame
+            The Pandas DataFrame containing the desired quantile for each component.
+        """
+        self._assert_stochastic()
+        raise_if_not(0 <= quantile <= 1,
+                     'The quantile values must be expressed as fraction (between 0 and 1 inclusive).', logger)
+
+        # column names
+        cnames = list(map(lambda s: s + '_{}'.format(quantile), self.columns))
+
+        return pd.DataFrame(self._xa.quantile(q=quantile, dim=DIMS[2]),
```

Comment:
I'm not 100% sure (or even 50% for that matter) how xarray.quantile works, but is this the intended behaviour? The sample=0 is usually not part of the "stochastic" TimeSeries, only the ones afterwards. Should we do something like self._xa[:,:,1:].quantile()?

Reply:
There is no difference between the sample 0 and the other samples. The only distinction is that "deterministic" series are defined to be those series that have exactly 1 sample, and the stochastic series those that have more than 1 samples. But the sample 0 is not special in any way in the stochastic series.